### PR TITLE
update 1-eks-cluster-with-new-vpc example with provider configuration

### PR DESCRIPTION
**Pull request by askulkarni2**
*Issue #, if available:* #156

*Description of changes:*

We are removing `providers.tf` as per module convention and updating our examples to show how consumer modules can provide us with their `provider {}` configurations. 

Tests:

Apply
```
terraform  apply -auto-approve
...
...
module.kubernetes-addons.module.aws_coredns[0].aws_eks_addon.coredns: Still creating... [15m0s elapsed]
module.kubernetes-addons.module.aws_coredns[0].aws_eks_addon.coredns: Creation complete after 15m7s [id=aws001-preprod-dev-eks:coredns]

Apply complete! Resources: 78 added, 0 changed, 0 destroyed.
```

Destroy
```
terraform destroy -auto-approve
...
...
module.aws_vpc.aws_vpc.this[0]: Destruction complete after 1s
module.aws-eks-accelerator-for-terraform.module.aws_eks.aws_iam_role.cluster[0]: Destruction complete after 2s

Destroy complete! Resources: 78 destroyed.
```  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kcoleman731** [comment]
@askulkarni2 don't you want to pass providers to the modules? I.E. here? https://github.com/aws-samples/aws-eks-accelerator-for-terraform/blob/task/update-examples-for-providers/deploy/1-eks-cluster-with-new-vpc/main.tf

**Comment by askulkarni2** [comment]
> @askulkarni2 don't you want to pass providers to the modules? I.E. here? https://github.com/aws-samples/aws-eks-accelerator-for-terraform/blob/task/update-examples-for-providers/deploy/1-eks-cluster-with-new-vpc/main.tf

@kcoleman731 doesn't look like we need to add that.

File: deploy/2-eks-cluster-with-import-vpc/eks/main.tf [hunk 1]
```diff
@@ -35,14 +35,39 @@ terraform {
   }
 }
 
+data "aws_region" "current" {}
+
+data "aws_availability_zones" "available" {}
+
+data "aws_eks_cluster" "cluster" {
+  name = module.aws-eks-accelerator-for-terraform.eks_cluster_id
+}
```
Comment (vara-bonthu) [inline deploy/2-eks-cluster-with-import-vpc/eks/main.tf:44]: I changed this file heavily with my new PR. May be you can revoke the change to this file.

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
One change requested just to avoid conflicts on next PR

**Comment by vara-bonthu** [comment]
> @askulkarni2 don't you want to pass providers to the modules? I.E. here? https://github.com/aws-samples/aws-eks-accelerator-for-terraform/blob/task/update-examples-for-providers/deploy/1-eks-cluster-with-new-vpc/main.tf

It's not required as per Terraform Docs. Provider config is only required in root module. All the sub modules will leverage the same provider defined in the root.

**Comment by kcoleman731** [comment]
Implicit Provider Inheritance https://www.terraform.io/language/modules/develop/providers#implicit-provider-inheritance

**Review by kcoleman731 — APPROVED** [review]

File: deploy/1-eks-cluster-with-new-vpc/main.tf [hunk 2]
```diff
@@ -142,4 +165,6 @@ module "kubernetes-addons" {
   enable_aws_load_balancer_controller = true
   enable_metrics_server               = true
   enable_cluster_autoscaler           = true
+
+  depends_on = [module.aws-eks-accelerator-for-terraform]
```
Comment (vara-bonthu) [inline deploy/1-eks-cluster-with-new-vpc/main.tf:169]: @askulkarni2 We may need to change this to this line based in our recent timing issues. `depends_on = [module.aws-eks-accelerator-for-terraform.managed_node_groups]`

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
One small change required and it can be merged after

**Pull request merged**